ls1intum/Artemis
# `Programming exercises`: Limit Jenkins builds to default branches

Close #5575 and close #4088

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises

### Motivation and Context



### Description


### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming exercise

1. As a student: change the branch to a random other one (NOT the default one) and commit and push your changes
2. Jenkins or Bamboo should NOT build the changes
3. Artemis should NOT have a submission and should NOT have a result for the submission on the wrong branch

### Review Progress





#### Code Review
#### Manual Tests
